micrometer-spring-legacy/src/main/java/io/micrometer/spring/autoconfigure/export/prometheus/PrometheusExportConfiguration.java

```diff
@@ -111,9 +130,17 @@ public PrometheusScrapeMvcEndpoint prometheusMvcEndpoint(PrometheusScrapeEndpoin
             this.pushgatewayProperties = prometheusProperties.getPushgateway();
             this.pushGateway = new PushGateway(pushgatewayProperties.getBaseUrl());
             this.environment = environment;
-            this.pushFuture = Executors.newSingleThreadScheduledExecutor()
-                    .scheduleAtFixedRate(this::push, pushgatewayProperties.getPushRate().toMillis(),
-                            pushgatewayProperties.getPushRate().toMillis(), TimeUnit.MILLISECONDS);
+            final ScheduledExecutorService executorService = Executors.newSingleThreadScheduledExecutor(
```

Comment:
Should the service be shutdown orderly when destroying app?

Reply:
It does so when the daemon thread dies. Arguably it would be cleaner to `shutdownNow()` just didn't wan't to change the status-quo.

Reply:
Just worried things can stick around when doing hot-reload.